### ISSUE TITLE
Blacks look gray with low brightness

### ISSUE THREAD

**Issue by borekb** [ojotoxy/Brightness#12]
I thought my Dell U3417W has terrible blacks that look gray, it was really quite unpleasant to work with the monitor during nights, at lower brightness setting. Only today I have found out that it's actually caused by this utility.

I don't know much about the technical details but what I observed is that similar utilities like [MonitorControl](https://github.com/the0neyouseek/MonitorControl) and [Lunar](https://github.com/alin23/Lunar) manipulate my monitor's brightness directly – I can see the value changing in OSD. With 'brightness', I don't see the value changing – is it using some kind of software overlay? Those two utilities maintain clear blacks.

I like 'brightness' overall – Lunar is quite complicated while MonitorControl doesn't sync with internal display – but the grey look is not bearable after seeing what the monitor can truly look like 😄. Is there a chance this could be fixed?

BTW my connection is via HDMI.

**Comment by borekb** [comment]
From the README (emphasis mine):

> **Two methods** are available for setting brightness, which I call **Real and Fake**.
>
> Real is supported on apple monitors. There turns out to be a semi-documented api for doing setting brightness. This stuff is sort of on the fringes of what is offically supported by Apple!
>
> **Fake** is supported on all monitors. This is **done using the graphics card Gamma Table**. Which basically maps RGB values to new RGB values. It is meant for color correction, but brightness correction is a subset of that. The original gamma table is read, and then scaled downward. (It does this the easy way now, but what might be better is to stretch the gamma table rightward, using interpolation. This could preserve color better while dimming? I thought I did this, but apparently it's doing it the dumb way...)

This is probably causing it but it's really quite unacceptable on my Dell monitor. Hopefully a new method could be found 😄.

Somewhat related: https://github.com/ojotoxy/Brightness/issues/9

**Comment by deepai-org** [comment]
Yes this is the difference between "Real" and "Fake" brightness as rambled about in the readme.
It is possible to add the real brightness control here. I just never found the time. Indeed I started once and never finished, copying some code from related projects.

"Real" brightness has some downsides though besides having better contrast. Monitor compatibiity issues. It can be unsure of which brightness level to use to match the macbook monitor. It can even crash the graphics driver needing you to reboot. 

That's all I got. Sorry.

**Comment by baragona** [comment]
Yes this is the difference between "Real" and "Fake" brightness as rambled about in the readme.
It is possible to add the real brightness control here. I just never found the time. Indeed I started once and never finished, copying some code from related projects.

"Real" brightness has some downsides though besides having better contrast. Monitor compatibiity issues. It can be unsure of which brightness level to use to match the macbook monitor. It can even crash the graphics driver needing you to reboot.

That's all I got. Sorry.

**Comment by lacek** [comment]
Used to use Brightness for around a year. I need to set the brightness and contrast to 100% to match that on Macbook.
It worked fine. Until recently I found that my LCD screen has temporary burn-in after several hours of use. So I switched to [Lunar](https://github.com/alin23/Lunar) which does the "real" brightness adjustment (at least on my Dell monitor). You may be interested in studying the source code for the mechanism behind.
